### ISSUE TITLE
[BUG] Wrong assumption regarding code location

### ISSUE THREAD

**Issue by mlopezFC** [vaadin/copilot#40]
### Description of the Bug

On certain projects, like add-ons, the source code of the demo is located on src/test/java instead of src/main/java. Copilot wrongly assumes that sources are on src/main/java when trying to interact with a view from a demo and throws a java.io.FileNotFoundException

### Expected Behavior

I would expect that if sources cannot be found on src/main/java, it should try with src/test/java

### Minimal Reproducible Example

Just try to use copilot on any add-on demo like [this one](https://github.com/FlowingCode/MarkdownEditor).

### Versions

- Vaadin Flow/Hilla Framework Version: e.g., 24.5.0

**Comment by abdullahtellioglu** [comment]
I created an example project with the addon you mentioned. Create location is not found - which is a bug- but it looks for the source directory, `src/main/java`. Full path of the component  `/copilot-client-test-flow/src/main/java/com/flowingcode/vaadin/addons/markdown/BaseMarkdownComponent.java`


This is the example code I created 
```java

    public HelloWorldView() {
        MarkdownEditor mde = new MarkdownEditor();
        mde.setSizeFull();
        mde.setPlaceholder("Enter Markdown here");
        mde.setMaxLength(500);
        mde.setDataColorMode(BaseMarkdownComponent.DataColorMode.LIGTH);
        add(mde);
    }
```
Could you provide more details about the project where Copilot locates the component in `src/test/java` folder?

**Comment by mlopezFC** [comment]
Sorry for the late response.
If instead of creating a new project with the add-on, you just clone [the add-on's repo](https://github.com/FlowingCode/MarkdownEditor), update the pom.xml file so it uses Vaadin 24.5 and start the demo with "mvn jetty:run", and activate copilot and click in the area highlighted by the purple cross, then you're going to see the exception telling that it can't find the class and it is trying to locate it in the "src/main/java" path instead of "src/test/java" path, where the MarkdownViewerDemo.java class is located.

![image](https://github.com/user-attachments/assets/61ba36eb-63aa-44bf-aaad-31f48cdada76)

**Comment by Artur-** [comment]
This was closed as fixed as 24.5.1 looks in the test folder also
